src/injected/analyzers/tab-stops-analyzer.ts

```diff
@@ -50,37 +47,14 @@ export class TabStopsAnalyzer extends BaseAnalyzer {
         };
         this.tabStopListenerRunner.start();
 
-        if (
-            this.tabStopRequirementRunner &&
-            this.featureFlagStore.getState()[FeatureFlags.tabStopsAutomation] === true
-        ) {
-            this.seenTabStopRequirementResults = [];
+        if (this.featureFlagStore.getState()[FeatureFlags.tabStopsAutomation] === true) {
             this.tabStopsDoneAnalyzingTracker.reset();
-            this.tabStopRequirementRunner.topWindowCallback = this.processTabStopRequirementResults;
-            this.tabStopRequirementRunner.start();
+            this.tabStopsRequirementResultProcessor.start(this.needsRequirementRunner);
```

Comment:
Why do this instead of the previous approach of checking for the null runner?

Reply:
I didn't really like having to create two `TabStopsRequirementResultProcessor`s in the MainWindowInitializer and passing them both into the AnalyzerProvider but if there's a better way to do it, I can change it

Reply:
I think another approach might be a wrapper/factory around the creation of it that the analyzer-provider can use.

Reply:
But I'm still not sure I understand why we can't use the null approach from before: I think we only need the `TabStopsRequirementResultProcessor` in TabStopsAnalyzer (and not in FocusTrackingAnalyzer).